### ISSUE TITLE
Expo Camera barcode scanner not working on WEB (Chrome)

### ISSUE THREAD

**Issue by t0ma5h** [expo/expo#29827]
### Minimal reproducible example

https://snack.expo.dev/@t0ma5h/expo-camera-repro

### What platform(s) does this occur on?

Web

### Did you reproduce this issue in a development build?

No (web-only issue)

### Summary

When implementing expo-camera on mobile seems to work out of the box with all types of barcodes. 
Web does not seem to scan any barcode (Tested with aztec, qr, and some others).

### Environment

 expo-env-info 1.2.0 environment info:
    System:
      OS: macOS 14.3.1
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 18.18.0 - /usr/local/bin/node
      Yarn: 1.22.19 - /usr/local/bin/yarn
      npm: 9.8.1 - /usr/local/bin/npm
    Managers:
      CocoaPods: 1.15.2 - /opt/homebrew/lib/ruby/gems/3.3.0/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.2, iOS 17.2, macOS 14.2, tvOS 17.2, visionOS 1.0, watchOS 10.2
    IDEs:
      Android Studio: 2022.3 AI-223.8836.35.2231.10671973
      Xcode: 15.2/15C500b - /usr/bin/xcodebuild
    npmPackages:
      expo: ~51.0.14 => 51.0.14 
      expo-router: ~3.5.16 => 3.5.16 
      react: 18.2.0 => 18.2.0 
      react-dom: 18.2.0 => 18.2.0 
      react-native: 0.74.2 => 0.74.2 
      react-native-web: ~0.19.6 => 0.19.12 
    npmGlobalPackages:
      eas-cli: 10.0.0
    Expo Workflow: managed

### Expo Doctor Diagnostics

✔ Check Expo config for common issues
✔ Check package.json for common issues
✔ Check native tooling versions
✔ Check dependencies for packages that should not be installed directly
✔ Check for common project setup issues
✔ Check for issues with metro config
✔ Check npm/ yarn versions
✔ Check that native modules do not use incompatible support packages
✔ Check for legacy global CLI installed locally
✔ Check Expo config (app.json/ app.config.js) schema
✔ Check that packages match versions required by installed Expo SDK
✔ Check that native modules use compatible support package versions for installed Expo SDK

Didn't find any issues with the project!

**Comment by smeckler** [comment]
I have the same problem: Could this be a camel-case mismatch between the props `onBarCodeScanned` and `onBarcodeScanned` which are both used?

**Comment by sagark1510** [comment]
same here! It works well on the mobile app but on the web it is not working and not scanning any QR code.

**Comment by sagark1510** [comment]
@t0ma5h did you find any solution?

**Comment by t0ma5h** [comment]
> @t0ma5h did you find any solution?

Not as of yet on SDK51. But just using sdk 50  for the time being should work

**Comment by entiendoNull** [comment]
> I have the same problem: Could this be a camel-case mismatch between the props `onBarCodeScanned` and `onBarcodeScanned` which are both used?

Yes, it seems like the current version of `camera` (SDK51) is using types from the `camera/legacy`. 

Some props have been renamed (such as `onBarCodeScanned` -> `onBarcodeScanned` and `type` -> `facing`) and simply no longer plays well with the current API.

This PR (originally there to fix the camera-facing-issue) has also targetted the onBarCodeScanned issue as a side-effect.

https://github.com/expo/expo/pull/30277

The current implementation only seem to work with QR-codes though. It should work if you feed the CameraView with `onBarcodeScanned` (and target QR-codes), but TypeScript will yell at you until this PR or any other fix is applied.

I believe this will get some attention pretty soon as the camera/legacy will be removed in SDK52 (if I remember correctly). In the meantime though, it would be good to get this patched up.

**Comment by henrilp** [comment]
I confirm that CameraView is not detecting qr codes on web (firefox).
I my experience, it was not working with previous BarCodeScanner package either.
I would be pleased for it to work :D thank you in advance

**Comment by shilo-klydo** [comment]
Same here

**Comment by matiasbattocchia** [comment]
Not working on web Firefox/Chrome, nor mobile web Firefox. Using expo@51.0.20 and expo-camera@15.0.16.

I'm calling the component like this

```js
      <CameraView
        barcodeScannerSettings={{
          barcodeTypes: ["qr"],
        }}
        onBarcodeScanned={onBarcodeScanned}
      >
         ...
      </CameraView>
```

**Comment by shiwofeiwo** [comment]
same issue CameraView onBarcodeScanned not working on web(chrome) (SDK51)

**Comment by Vincenzo1301** [comment]
Same issue. Barcodes are not scanned on web (chrome)

```
 const handleBarcodeScanned = ({type, data}: BarCodeScanningResult) => {
        console.log(`Bar code with type ${type} and data ${data} has been scanned!`);
        setScanned(true);
        alert(`Bar code with type ${type} and data ${data} has been scanned!`);
    };

<CameraView
                barcodeScannerSettings={{
                    barcodeTypes: ["ean13", "code128", "qr"],
                }}
                onBarcodeScanned={scanned ? undefined : handleBarcodeScanned}
                facing={cameraType}
                enableTorch={flashOn}
                style={StyleSheet.absoluteFillObject}
            />
```

**Comment by kenta3434** [comment]
Same issue: CameraView ean13 not working on iOS, but QR code scanning worked successfully

**Comment by DaliaaDawod** [comment]
Anyone find a solution for this?
The expo-camera works on ios for me but not for web.
I have:
 "expo": "^51.0.10",
 "expo-camera": "~15.0.16",

**Comment by xposionn** [comment]
Any update on this ?

**Comment by pianiomar** [comment]
still only QR works on WEB with 

```
"expo": "~52.0.18",
"expo-camera": "~16.0.10",
```

is there an older configuration working with "EAN8/13" in web?

**Comment by clintwarner** [comment]
Would love for a useable solution. Seems like all other barcode scanning packages require a polyfill that use a web assembly, which Expo doesn't allow........ So if we use Expo we are locked into to using their camera, which has been known to be broken for nearly a year now

**Comment by sergeushenecz** [comment]
> "expo": "~52.0.18",
> "expo-camera": "~16.0.10",

doesnt and qr code for me.

Only works qr code on 
"expo": "51.0.39",
 "expo-camera": "15.0.16",

**Comment by sergeushenecz** [comment]
For expo 52 version and expo-camera 16 version need use for web 
not this code 
```
barCodeScannerSettings: {
      barCodeTypes: ["qr"]
    }

```

barcodeScannerSettings: {
      barcodeTypes: ["qr"]
    }


I've found it here in source code for web

<img width="1243" alt="Image" src="https://github.com/user-attachments/assets/c6f09985-675b-4e94-9f1c-d66e85bd0d6b" />

**Comment by AJZander** [comment]
If anyone has tried with the new expo SDK 53 and had success with this I would love to know?

**Comment by 2winter-dev** [comment]
When I start scanning a QR code on Android, the Google QR code scanner will be opened. Is there a way to not use it?

**Comment by syrenius** [comment]
> If anyone has tried with the new expo SDK 53 and had success with this I would love to know?

Still not working for me with SDK 53 😕

One thing I noticed: all my webcams produce a mirrored image, and it seems like the mirror property isn’t being respected. Could this be preventing the codes from being recognized? I did a quick test by printing a mirrored Code128 barcode, but unfortunately, I still don’t get any onBarcodeScanned event...

**Comment by Randall71** [comment]
@entiendoNull could you push this one now for a fix  ?

**Comment by creambyemute** [comment]
On Expo 54 (54.0.12) with Camera 17.0.8 it works when adding `barcodeScannerSettings={{ barcodeTypes: ["qr"] }}` to the CameraView.

If I don't add it it doesn't work on web (but it does for iOS/Android)

**Comment by sergeushenecz** [comment]
> On Expo 54 (54.0.12) with Camera 17.0.8 it works when adding `barcodeScannerSettings={{ barcodeTypes: ["qr"] }}` to the CameraView.
> 
> If I don't add it it doesn't work on web (but it does for iOS/Android)

@creambyemute  For code 128 support on web or no ?
